staramr/blast/JobHandler.py

```diff
@@ -62,13 +63,13 @@ def __init__(self, blast_database_objects_map: Dict[str, AbstractBlastDatabase],
         else:
             self._pointfinder_configured = True  # type: bool
 
-        self._thread_pool_executor = None
+        self._thread_pool_executor = ThreadPoolExecutor(max_workers=self._threads)
```

Comment:
I tried doing that initially but for some reason it gives me this following error when I do that:

```
2019-07-23 09:32:50 ERROR staramr.<module>,75: 'JobHandler' object has no attribute '_thread_pool_executor'
Traceback (most recent call last):
  File "jtran/Projects/staramr/bin/staramr", line 68, in <module>
    args.run_command(args)
  File "jtran/Projects/staramr/staramr/subcommand/Search.py", line 431, in run
    ignore_invalid_files=args.ignore_valid_files)
  File "jtran/Projects/staramr/staramr/subcommand/Search.py", line 244, in _generate_results
    'plasmidfinder': plasmidfinder_database}, nprocs, blast_out)
  File "jtran/Projects/staramr/staramr/blast/JobHandler.py", line 68, in __init__
    self.reset()
  File "jtran/Projects/staramr/staramr/blast/JobHandler.py", line 75, in reset
    if self._thread_pool_executor is not None:
AttributeError: 'JobHandler' object has no attribute '_thread_pool_executor'
```

Upon looking into this further I realized it no longer gives me an error when I initialized it to None. So I'll be reverting it back to the way it initially was.

Reply:
If it's giving you a warning, then I would keep the code you have. I thought maybe it would be okay only setting it in `reset()`, but I guess not. Thanks for looking into this :smile:

Reply:
No need, the warning is mainly a type definition. So I was still able to remove it completely without any issues